keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
+## BucketRequest
 
-This call is made to create the bucket in the backend. If a bucket that matches both name and parameters already exists, then OK (success) must be returned. If a bucket by same name, but different parameters is provided, then the appropriate error code `ALREADY_EXISTS` must be returned by the provisioner. The call to *ProvisonerCreateBucket* MUST be idempotent.
+A request to create a Bucket in OSP. BucketRequest is a namespaced resource
 
+```yaml
+BucketRequest {
+  TypeMeta
+  ObjectMeta
+  
+  Spec BucketRequestSpec {
+    // Protocol is the data API this bucket is expected to adhere to.
+    // The possible values for protocol are:
+    // -  S3: Indicates Amazon S3 protocol
+    // -  Azure: Indicates Microsoft Azure BlobStore protocol
+    // -  GCS: Indicates Google Cloud Storage protocol
+    Protocol Protocol
+    
+    // Name of the BucketClass 
+    // +optional    
+    BucketClassName string
+  }
+    
+  Status BucketRequestStatus {
+    // BucketAvailable indicates that the bucket is ready for consumpotion
+    // by workloads      
+    BucketAvailable bool 
+  }
 ```
-message ProvisionerCreateBucketRequest {    
-    // This field is REQUIRED
-    string bucket_name = 1;
+## BucketClass
 
-    map<string,string> bucket_context = 2;
+Resouce for configuring common properties for multiple Buckets. BucketClass is a clustered resource
 
-    enum AnonymousBucketAccessMode {
-	PRIVATE = 0;
-	PUBLIC_READ_ONLY = 1;
-	PUBLIC_WRITE_ONLY = 2;
-	PUBLIC_READ_WRITE = 3;
-    }
-    
-    AnonymousBucketAccessMode anonymous_bucket_access_mode = 3;
+```yaml
+BucketClass {
+  TypeMeta
+  ObjectMeta
+  
+  // Provisioner is the name of driver associated with this bucket
+  Provisioner string
+  
+  // DeletionPolicy is used to specify how COSI should handle deletion of this 
+  // bucket. There are 3 possible values:
+  //  - Retain: Indicates that the bucket should not be deleted from the OSP
+  //  - Delete: Indicates that the bucket should be deleted from the OSP
+  //        once all the workloads accessing this bucket are done
+  //  - ForceDelete: Indicates that the bucket and its contents should be 
+  //        deleted from the OSP even if workloads are currently accessing 
+  //        this bucket
+  DeletionPolicy DeletionPolicy
+
+  // AllowedNamespaces is the list of namespaces from which workloads     
+  // can access this Bucket. If empty, all namespaces can access this access
+  // +optional   
+  AllowedNamespaces []string
+
+  // Parameters is an opaque map for passing in configuration to a driver
+  // for creating the bucket
+  // +optional
+  Parameters map[string]string 
 }
+```
 
-message ProvisionerCreateBucketResponse {
-    // Intentionally left blank
+## BucketAccess
+
+```yaml
+BucketAccess {
+  TypeMeta
+  ObjectMeta
+  
+  Spec BucketAccessSpec {
+    // Parameters is an opaque map for passing in configuration to a driver
+    // for creating the bucket
+    // +optional
```

Comment:
How can this field be optional? Without the BAR there is no way to reference the B (via  BAR.spec.BR).

Reply:
good catch. Will fix it